crates/neon-runtime/src/napi/tag.rs

```diff
@@ -1,23 +1,40 @@
-use raw::Local;
+use raw::{Env, Local};
 
-pub unsafe extern "C" fn is_undefined(_val: Local) -> bool { unimplemented!() }
+use nodejs_sys as napi;
 
-pub unsafe extern "C" fn is_null(_val: Local) -> bool { unimplemented!() }
+unsafe fn is_type(env: Env, val: Local, expect: napi::napi_valuetype) -> bool {
+    let mut actual = napi::napi_valuetype::napi_undefined;
+    if napi::napi_typeof(env, val, &mut actual as *mut _) == napi::napi_status::napi_ok {
+        actual == expect
+    } else {
+        false
+    }
+}
 
-pub unsafe extern "C" fn is_number(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_undefined(_env: Env, _val: Local) -> bool { unimplemented!() }
 
-pub unsafe extern "C" fn is_boolean(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_null(_env: Env, _val: Local) -> bool { unimplemented!() }
 
-pub unsafe extern "C" fn is_string(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_number(env: Env, val: Local) -> bool {
+    is_type(env, val, napi::napi_valuetype::napi_number)
+}
 
-pub unsafe extern "C" fn is_object(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_boolean(env: Env, val: Local) -> bool {
+    is_type(env, val, napi::napi_valuetype::napi_boolean)
+}
 
-pub unsafe extern "C" fn is_array(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_string(env: Env, val: Local) -> bool {
+    is_type(env, val, napi::napi_valuetype::napi_string)
+}
 
-pub unsafe extern "C" fn is_function(_val: Local) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_object(_env: Env, _val: Local) -> bool { unimplemented!() }
```

Comment:
Any reason you didn't implement this one yet? Not a big deal, we can take it incrementally, just curious.

Reply:
Since it wasn't the focus of the PR, I only implemented the ones that were required to make the tests work. #507 includes a few more already